sklearn/linear_model/_bayes.py

```diff
@@ -228,6 +246,23 @@ def fit(self, X, y, sample_weight=None):
         """
         self._validate_params()
 
+        max_iter = self.max_iter
+        # TODO(1.5) Remove
+        if self.n_iter != "deprecated":
+            if max_iter is not None:
+                raise ValueError(
+                    "Both `n_iter` and `max_iter` attributes were set. Attribute"
+                    " `n_iter` was deprecated in version 1.3 and will be removed in"
+                    " 1.5. To avoid this error, only set the `max_iter` attribute."
+                )
+            warnings.warn(
+                "'n_iter' was renamed to 'max_iter' in version 1.3 and "
+                "will be removed in 1.5",
+                FutureWarning,
+            )
+            max_iter = self.n_iter
+        elif max_iter is None:
+            max_iter = 300
```

Comment:
Let's create a small function called `_deprecate_max_iter` that take an estimator and return `max_iter`. Like this we can use it in both class without code duplication.

Reply:
I'd say in `_bayes.py` just to keep it local